### ISSUE TITLE
Outdated info in 'Add support for variables' topic

### ISSUE THREAD

**Issue by josmperez** [grafana/plugin-tools#829]
Reported by Andriy Urbanas:

1. There are some documentation sections that are outdated. 

Example: [Add support for variables](https://grafana.com/developers/plugin-tools/create-a-plugin/extend-a-plugin/add-support-for-variables#add-support-for-query-variables-to-your-data-source). At the end of the page it mentions using setVariableQueryEditor  for configuring plugin's query editor for variables. But actually it's deprecated (you get deprecation message when you start using it and it mentions the new way) and there is a new way of adding variables, which is not documented. Once you go to the code to understand this new method - it's also not clear, methods / properties are not documented, comments are not clear. So you need to check some examples and play around with it until you get something you want. 

There is a [slack thread](https://raintank-corp.slack.com/archives/C03KVDHTWAH/p1706878535598039) I started a while ago, it has some more info.

2. Incorrect docs. For example on the same page about [adding support for variables](https://grafana.com/developers/plugin-tools/create-a-plugin/extend-a-plugin/add-support-for-variables#add-support-for-query-variables-to-your-data-source), there is this:

The metricFindQuery function returns an array of MetricFindValue which has a single property, text:
async metricFindQuery(query: MyVariableQuery, options?: any) {
  // Retrieve DataQueryResponse based on query.
  const response = await this.fetchMetricNames(query.namespace, query.rawQuery);

  // Convert query results to a MetricFindValue[]
  const values = response.data.map(frame => ({ text: frame.name }));

  return values;
}

But in reality MetricFindValue has not only text, it also has value  and expandable :
interface MetricFindValue {
    text: string;
    value?: string | number;
    expandable?: boolean;
}

**Comment by zoltanbedi** [comment]
I just bumped into this. Are you sure you want to leave the deprecated api in the docs?

**Comment by josmperez** [comment]
> I just bumped into this. Are you sure you want to leave the deprecated api in the docs?

Not at all, thanks for calling this to my attention. I'm looking into it.

**Comment by sympatheticmoose** [comment]
Related https://github.com/grafana/grafana-plugin-examples/issues/141

**Comment by mckn** [comment]
I'm happy to help out with this @josmperez. Give me a ping and I can show case the "new" approach to this.
